### PR TITLE
feat: add user logs

### PR DESCRIPTION
**Pull request by voikin**
more details in `ASP-712`

### CODE REVIEW

File: controllers/role.go [hunk 1]
```diff
@@ -101,8 +102,55 @@ func (c *ApiController) UpdateRole() {
...
+			oldUsers[userID] = struct{}{}
+		}
+
+		for _, userID := range role.Users {
+			newUsers[userID] = struct{}{}
+		}
+
+		for userID := range oldUsers {
+			if _, found := newUsers[userID]; !found {
+				logger.Info(ctx, "user lost role",
```
Comment (svgrafov) [inline controllers/role.go:136]: I believe we can make this message more clear to something like:"Role removed from user". It seems like the user did nothing to lose the role, so he is not an actor of this event.

File: controllers/role.go [hunk 2]
```diff
@@ -101,8 +102,55 @@ func (c *ApiController) UpdateRole() {
...
+					"role_id", role.GetId(),
+					"user_id", userID,
+					"by_user", c.getCurrentUser().GetId())
+			}
+		}
+
+		for userID := range newUsers {
+			if _, found := oldUsers[userID]; !found {
+				logger.Info(ctx, "user gained role",
```
Comment (svgrafov) [inline controllers/role.go:145]: And this one can become "role added to user" for uniformity.

File: controllers/user.go [hunk 3]
```diff
@@ -313,6 +314,8 @@ func (c *ApiController) UpdateUser() {
 	var user object.User
 	err := json.Unmarshal(c.Ctx.Input.RequestBody, &user)
 	if err != nil {
+		logger.Error(goCtx, "failed to unmarshall user",
```
Comment (svgrafov) [inline controllers/user.go:317]: does this message have operation(`UpdateUser()`) name added under the hood? 
If not, we can change this to 

`logger.Error(goCtx, errors.Wrap(err, "UpdateUser failed: failed to unmarshal request body").Error())`

File: controllers/user.go [hunk 4]
```diff
@@ -322,41 +325,59 @@ func (c *ApiController) UpdateUser() {
 	if id == "" {
 		id = c.GetSessionUsername()
 		if id == "" {
+			logger.Error(goCtx, "missed parameter to update user",
```
Comment (svgrafov) [inline controllers/user.go:328]: Same concern about operation name applies here.
We also need to add name of that missing parameter to error message.

File: controllers/user.go [hunk 5]
```diff
@@ -322,41 +325,59 @@ func (c *ApiController) UpdateUser() {
...
 
 	if oldUser.Owner == "built-in" && oldUser.Name == "admin" && (user.Owner != "built-in" || user.Name != "admin") {
+		logger.Error(goCtx, "tried to update admin user")
 		c.ResponseForbidden(c.T("auth:Unauthorized operation"))
 		return
 	}
 
 	if c.Input().Get("allowEmpty") == "" {
 		if user.DisplayName == "" {
+			logger.Error(goCtx, "tried to set empty display name",
```
Comment (svgrafov) [inline controllers/user.go:359]: I belive this message needs more clarity too.
It can become smth like:"UpdateUser failed: empty display name not allowed"

**Review by svgrafov — CHANGES_REQUESTED** [review]
Please add operation name to errors to make them more searchable when looking over logs.

File: controllers/role.go (hunk 1 above)
Comment (voikin) [reply controllers/role.go:136]: done

File: controllers/role.go (hunk 2 above)
Comment (voikin) [reply controllers/role.go:145]: done

File: controllers/user.go (hunk 3 above)
Comment (voikin) [reply controllers/user.go]: done

File: controllers/user.go (hunk 4 above)
Comment (voikin) [reply controllers/user.go]: done

File: controllers/user.go (hunk 5 above)
Comment (voikin) [reply controllers/user.go]: done

**Review by svgrafov — APPROVED** [review]

**Review by dplynskiy — APPROVED** [review]

**Review by svgrafov — APPROVED** [review]

**Pull request merged**